contracts/staking/contracts/src/staking_pools/MixinStakingPoolRewards.sol

```diff
@@ -17,297 +17,225 @@
+        IStructs.Fraction memory mostRecentCumulativeRewards = cumulativeRewardsByPoolPtr[cumulativeRewardsLastStored];
+
+        // compute new cumulative reward
+        (uint256 numerator, uint256 denominator) = LibSafeMath.addFractions(
+            mostRecentCumulativeRewards.numerator,
+            mostRecentCumulativeRewards.denominator,
+            reward,
+            amountOfDelegatedStake
         );
 
-        // the buy-in will be > 0 iff there exists a non-zero reward.
-        if (buyIn > 0) {
-            shadowRewardsInPoolByOwner[member][poolId] = shadowRewardsInPoolByOwner[member][poolId].safeAdd(buyIn);
-            shadowRewardsByPoolId[poolId] = shadowRewardsByPoolId[poolId].safeAdd(buyIn);
-        }
+        // normalize fraction components by dividing by the min token value (10^18)
+        (uint256 numeratorNormalized, uint256 denominatorNormalized) = (
+            numerator.safeDiv(MIN_TOKEN_VALUE),
+            denominator.safeDiv(MIN_TOKEN_VALUE)
+        );
+
+        // store cumulative rewards
+        cumulativeRewardsByPoolPtr[epoch] = IStructs.Fraction({
+            numerator: numeratorNormalized,
+            denominator: denominatorNormalized
+        });
+        cumulativeRewardsByPoolLastStored[poolId] = epoch;
     }
 
-    /// @dev A member leaves a staking pool.
-    /// This function decrements the shadow balance of the member, along
-    /// with the total shadow balance of the pool. This ensures that
-    /// any rewards belonging to co-members will not be inflated.
-    /// @param poolId Unique Id of pool to leave.
-    /// @param member The member to leave.
-    /// @param amountOfStakeToUndelegate The stake to be undelegated by `member` upon leaving.
-    /// @param totalStakeDelegatedToPoolByMember The amount of stake currently delegated to the pool by the member.
-    ///                                          This includes `amountOfStakeToUndelegate`.
-    /// @param totalStakeDelegatedToPool The total amount of stake currently delegated to the pool, across all members.
-    ///                                  This includes `amountOfStakeToUndelegate`.
-    function _leaveStakingPool(
+    /// @dev Computes a member's reward over a given epoch interval.
+    /// @param poolId Uniqud Id of pool.
+    /// @param memberStakeOverInterval Stake delegated to pool by meber over the interval.
+    /// @param beginEpoch beginning of interval.
+    /// @param endEpoch end of interval.
+    /// @return rewards accumulated over interval [beginEpoch, endEpoch]
+    function _computeMemberRewardOverInterval(
         bytes32 poolId,
-        address payable member,
-        uint256 amountOfStakeToUndelegate,
-        uint256 totalStakeDelegatedToPoolByMember,
-        uint256 totalStakeDelegatedToPool
+        uint256 memberStakeOverInterval,
+        uint256 beginEpoch,
+        uint256 endEpoch
     )
-        internal
+        private
```

Comment:
I understand why this is being made private, but it may be wise to leave this as `internal` for the purposes of unit testing. Thoughts?

Reply:
Yeah, I think you're right. For unit testing these will have to be made internal =\